c7n/resources/iam.py

```diff
@@ -817,6 +817,41 @@ def process(self, resources):
                     pass
 
 
+@Role.action_registry.register('delete')
+class RoleDelete(BaseAction):
+    """Delete an IAM Role.
+
+    For example, if you want to automatically delete an unused IAM role.
+
+    :example:
+
+      .. code-block:: yaml
+
+        - name: iam-delete-unused-role
+          resource: iam-role
+          filters:
+            - type: unused
+          actions:
+            - delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ('iam:DeleteRole',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        for r in resources:
+            try:
+                client.delete_role(RoleName=r['RoleName'])
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'DeleteConflict':
```

Comment:
this needs to raise after processing the remainder.

Reply:
Added code to raise at the end